examples/local_damage.py

```diff
@@ -0,0 +1,249 @@
+        """
+        Postprocessing class to track the tractions at `boundary` (actually 
+        tractions . direction) and the displacements at the boundary.
+
+        Each update to the plot can visualized using 
+            fenics_helpers.plotting.AdaptivePlot
+        """
+        mesh = model.mesh
+        self.model = model
+        boundary_markers = df.MeshFunction("size_t", mesh, mesh.topology().dim() - 1)
+
+        boundary.mark(boundary_markers, marker)
+        ds = df.Measure("ds", domain=mesh, subdomain_data=boundary_markers)
+
+        n = df.FacetNormal(mesh)
+        if direction is None:
+            direction = n
+        self.boundary_load = df.dot(model.traction(n), direction) * ds(marker)
+        self.boundary_disp = df.dot(model.d, direction) * ds(marker)
+        self.area = df.assemble(1.0 * ds(marker))
+
+        self.load = []
+        self.disp = []
+
+        self.plot = None
+
+    def __call__(self, t):
+        load = df.assemble(self.boundary_load)
+        disp = df.assemble(self.boundary_disp) / self.area
+        self.load.append(load)
+        self.disp.append(disp)
+        if self.plot is not None:
+            self.plot(disp, load)
+
+    def show(self, fmt="-rx"):
+        self.plot = fh.plotting.AdaptivePlot(fmt)
+
+class Plotter:
+    def __init__(self, model, filename=None):
+        self.model = model
+        if filename is None:
+            filename = "out.xdmf"
+        self.plot = df.XDMFFile(filename)
+        self.plot.parameters["functions_share_mesh"] = True
+        
+        self.model.d.rename("u", "u")
+        self.model.k.rename("kappa", "kappa")
+
+    def __call__(self, t):
+        self.plot.write(self.model.d, t)
+        self.plot.write(self.model.k, t)
+
+problem = Problem(Mat(), l_panel_mesh(10, 10, refinement=4), order=1)
+
+# =====================================================
+#   Select boundaries via the fh.boundary module
+# =====================================================
+
+bot = fh.boundary.plane_at(-10, "y") 
+right = fh.boundary.plane_at(10, "x") 
+
+bc_top_expr=df.Expression("du * t", du=0.5, t=0, degree=0)
+
+bc_bot = df.DirichletBC(problem.W, [0,0], bot)
+bc_top_x = df.DirichletBC(problem.W.sub(0), 0, right)
+bc_top_y = df.DirichletBC(problem.W.sub(1), bc_top_expr, right)
+
+ld = LoadDisplacementCurve(problem, right, direction=df.Constant((0, 1)))
+ld.show()
+ld.plot.ax.set_ylabel("load [N]")
+ld.plot.ax.set_xlabel("displacement [mm]")
+
+plot_2d = Plotter(problem)
+
+solver = problem.get_solver([bc_bot, bc_top_y])
+
+# =====================================================
+#   Define solve() and pp() to use by the 
+#     fh.timestepping module
+# =====================================================
+
+def solve(t, dt):
+    bc_top_expr.t=t
+    return solver.solve()
+
+def pp(t):
+    problem.update()
+    ld(t)
+    plot_2d(t)
+
+ts = fh.timestepping.TimeStepper(solve, pp, u=problem.d)
+ts.increase_num_iter = 7
+ts.adaptive(1)
```

Comment:
Maybe say `ts.adaptive(t_end=1.)` to be explicit and avoid the magic number. (With it being an integer, I really didn't know what it was supposed to be.)